src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +241,101 @@ export class QuerySummary extends Component {
     }
   }
 
+  private parseNoResultsFoundMessage(noResultsFoundMessage: string) {
+    if (this.isQuerySummaryTagInMessage(noResultsFoundMessage)) {
+      let messageSections = noResultsFoundMessage.split(QUERY_TAG);
+      let messageElements = Array<Dom>();
+
+      _.each(messageSections, section => {
+        messageElements.push(this.createTextElement(section));
+      });
+
+      for (let i = 0, j = messageElements.length - 1; i < j; i++) {
```

Comment:
The method name does not really match what the method is performing. The name implies that it is only "parsing" the string, but here we are modifying the message HTML element.

I would move this processing in a separate method.

I think it's okay since we are escape the query element in `this.getQueryElement()`, but we need to make 100% sure it's correct, with a UT to make sure we are not vulnerable to any XSS attack. It's extremely important :)